app/controllers/devise_token_auth/sessions_controller.rb

```diff
@@ -55,6 +55,16 @@ def destroy
         user.tokens.delete(client)
         user.save!
 
+        if DeviseTokenAuth.cookie_enabled
+          if DeviseTokenAuth.cookie_attributes[:domain]
```

Comment:
I just realized that we might need to be smarter about the `domain` attribute. We might need to determine it by looking at the request's domain (in which case we'd probably want to allow configuring a whitelist, like we do with `redirect_whitelist`). This is because the `Domain` cookie attribute only allows one value, so as the code sits now you wouldn't be able to handle multiple domains being served from the same box.

Reply:
Make sense, remember to use allowlist instead of whitelist

Reply:
@apuntovanini Another option is to just support passing to the `cookie_attributes[:domain]` config either a string, which will work for most people, or a proc, which could accept the request and do whatever logic (in our case, return the domain of the request so long as it matches the allow list).

Reply:
[Here](https://github.com/lynndylanhurley/devise_token_auth/pull/1453/commits/7592336caa29f8a1c4d9173f409993becba6e1ef) are the changes for that second strategy. And my config looks like:

```
    config.cookie_attributes = {
        domain: lambda { |request| request.domain if valid_cookie_domains.include?(request.domain) },
        secure: Rails.env.production?,
        httponly: true,
        same_site: 'Strict',
        expires: token_lifespan.from_now,
        encrypt: true
    }
```

Reply:
🤦  I just saw this in the [in the Rails docs](https://api.rubyonrails.org/classes/ActionDispatch/Cookies.html):

> To support multiple domains, provide an array, and the first domain matching request.host will be used. Make sure to specify the :domain option with :all or Array again when deleting cookies.

So I can totally gut this Proc logic now, and instead just pass along either a string or Array from the config to the cookie call.

Reply:
Actually, using an array doesn't quite work for us since the `request.host` that it matches against includes the subdomain. But I think the `:all` option will:

> If you use a schema like www.example.com and want to share session with user.example.com set :domain to :all

Reply:
See [this commit](https://github.com/lynndylanhurley/devise_token_auth/pull/1453/commits/66f8fa98cfbefbeaa5f098f3efa482744f9e668d). So much cleaner! I wish I had realized that earlier, lol.